usr/share/rear/rescue/default/500_ssh.sh

```diff
@@ -54,5 +57,110 @@ if has_binary sshd; then
+                        LogUserOutput ""
+
+                        while true; do
+                            LogUserOutput "SSH client passphrase (or empty, or 'UnProtected'): \c"
+                            read -s passphrase <&6
+                            UserOutput ""
+
+                            case "$passphrase" in
+                                "")
+                                    LogUserOutput "SSH private keys will not be copied"
+                                    break
+                                    ;;
+
+                                UnProtected)
+                                    LogUserOutput ""
+                                    LogUserOutput "WARNING: This will protentially allow anyone in possession of the rescue medium"
+                                    LogUserOutput "unauthenticated access to other systems."
+                                    LogUserOutput ""
+                                    user_input="$( UserInput -I SSH_PRIVATE_KEYS_UNPROTECTED_CONFIRMATION \
+                                        -t 0 -p "Type 'yes' if you want to allow unauthenticated access" )"
+                                    if [ "$user_input" == "yes" ]; then
+                                        LogUserOutput "SSH private keys will be copied in unprotected form"
+                                        break
+                                    fi
+                                    ;;
+
+                                *)
+                                    LogUserOutput "Repeat SSH client passphrase: \c"
+                                    read -s repeated_passphrase <&6
+                                    UserOutput ""
+                                    if [ "$passphrase" != "$repeated_passphrase" ]; then
+                                        LogUserOutput "Passphrases do not match"
+                                        passphrase=""
+                                    else
+                                        LogUserOutput "SSH private keys will be copied and passphrase-protected"
+                                        break
+                                    fi
+                                    ;;
+                            esac
+                        done
+                    fi
+
+                    # Carry out the user's decision
+                    case "$passphrase" in
+                        "")
+                            Log "Ignoring unprotected private key $original_file"
+                            rm "$target_file"
+                            ;;
+
+                        UnProtected)
+                            Log "Copying unprotected private key $original_file"
+                            ;;
+
+                        *)
+                            Log "Copying private key $original_file, adding passphrase-protection"
+                            if ! ssh-keygen -p -P '' -N "$passphrase" -f "$target_file"; then
```

Comment:
If you  change the case to a nested if then you can copy the file here. Pseudocode like this:
```bash
if test "$passphrase" ; then
  cp $v -p "$original_file" "$target_dir" 1>&2 || Error "Could not copy $original_file to $target_dir"
  if [ "$passphrase" != UnProtected ] ; then
    ssh-keygen -p -P '' -N "$passphrase" -f "$target_file" || Error "Could not set passphrase on SSH key $target_file"
  fi
fi
```

IMHO this reads much shorter and more straight-forward.

Reply:
W.r.t. copying, see above.

I don't mind the short form for the error case, but I'd rather keep a log entry for each private key handled.

Reply:
Log entry is fine, I wouldn't bother the user in the verbose output for stuff that "just works".